Crashes when casting to a device.

**App version:** 1.7 (from Google Play)

**Android version**: 8.1

**Device model**: x4

**Expected behaviour**:
Casting to a device works normally

**Current behaviour**:
App crashes when casting to a device, then hitting play

**First occured**: Version 1.7 

**Steps to reproduce**:

Overflow menu > play on > select device > play button > App crashes (this used to work fine in version 1.6 and earlier)

Play button > Cast to > select device > works normally and the podcast plays on the selected device

```
java.lang.RuntimeException: Unable to create service de.danoeh.antennapod.core.service.playback.PlaybackService: java.lang.NullPointerException: Attempt to invoke virtual method 'int de.danoeh.antennapod.core.service.playback.PlaybackServiceMediaPlayer.getPosition()' on a null object reference
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:2887)
	at android.app.ActivityThread.access$1900(ActivityThread.java:150)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1427)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5417)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int de.danoeh.antennapod.core.service.playback.PlaybackServiceMediaPlayer.getPosition()' on a null object reference
	at de.danoeh.antennapod.core.service.playback.PlaybackService.access$400(PlaybackService.java:16107)
	at de.danoeh.antennapod.core.service.playback.PlaybackService$2.statusChanged(PlaybackService.java:669)
	at de.danoeh.antennapod.core.service.playback.PlaybackServiceMediaPlayer.setPlayerStatus(PlaybackServiceMediaPlayer.java:326)
	at de.danoeh.antennapod.core.service.playback.RemotePSMP.onRemoteMediaPlayerStatusUpdated(RemotePSMP.java:224)
	at de.danoeh.antennapod.core.service.playback.RemotePSMP.<init>(RemotePSMP.java:68)
	at de.danoeh.antennapod.core.service.playback.PlaybackServiceFlavorHelper.onCastAppConnected(PlaybackServiceFlavorHelper.java:178)
	at de.danoeh.antennapod.core.service.playback.PlaybackService.onCreate(PlaybackService.java:4068)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:2877)
	... 8 more
```

--- a/core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java
@@ -1096,7 +1096,7 @@ public class PlaybackService extends MediaBrowserServiceCompat {
         } else {
             state = PlaybackStateCompat.STATE_NONE;
         }
-        sessionState.setState(state, mediaPlayer.getPosition(), mediaPlayer.getPlaybackSpeed());
+        sessionState.setState(state, getCurrentPosition(), getCurrentPlaybackSpeed());
         long capabilities = PlaybackStateCompat.ACTION_PLAY_PAUSE
                 | PlaybackStateCompat.ACTION_REWIND
                 | PlaybackStateCompat.ACTION_FAST_FORWARD
@@ -1636,6 +1636,9 @@ public class PlaybackService extends MediaBrowserServiceCompat {
     }
 
     public float getCurrentPlaybackSpeed() {
+        if(mediaPlayer == null) {
+            return 1.0f;
+        }
         return mediaPlayer.getPlaybackSpeed();
     }
 

